deploy/eck-stack/charts/eck-fleet-server/examples/fleet-server.yaml

```diff
@@ -0,0 +1,19 @@
+version: 8.17.0-SNAPSHOT
+deployment:
+  replicas: 1
+  podTemplate:
+    spec:
+      serviceAccountName: fleet-server
+      automountServiceAccountToken: true
+elasticsearchRefs:
+- name: eck-elasticsearch
+  namespace: default
+kibanaRef:
+  name: eck-kibana
+  namespace: default
+http:
+  service:
+    spec:
+      type: ClusterIP
+serviceAccount:
+  name: elastic-fleet-server
```

Comment:
![image](https://github.com/user-attachments/assets/87bf46e3-8bdd-460a-8436-1ae568807dda)
All of the other charts expect `serviceAccount.name`, but render in the template `serviceAccountName`.

Reply:
```
find . -name 'values.yaml' -exec grep -H -A 1 '^# serviceAccoun' \{\} \;
./eck-stack/charts/eck-kibana/values.yaml:# serviceAccountName: ""
./eck-stack/charts/eck-kibana/values.yaml-
./eck-stack/charts/eck-elasticsearch/values.yaml:# serviceAccountName: ""
./eck-stack/charts/eck-elasticsearch/values.yaml-
./eck-stack/charts/eck-apm-server/values.yaml:# serviceAccountName: ""
./eck-stack/charts/eck-apm-server/values.yaml-
./eck-stack/charts/eck-logstash/values.yaml:# serviceAccountName: ""
./eck-stack/charts/eck-logstash/values.yaml-
./eck-stack/charts/eck-enterprise-search/values.yaml:# serviceAccountName: ""
./eck-stack/charts/eck-enterprise-search/values.yaml-
```

Reply:
Well, I'll make fleet consistent here, and then open an additional pr for beats/agent that's backwards compatible to make them also consistent.

Reply:
I will try to take another look to understand better why we did that.

Reply:
After looking throught the git history I could not find a good argument why we structured `serviceAccount` property in the Agent and Beats charts in the way we did. 

In all the other charts `serviceAccountName` serves only the [cross-namespace RBAC feature](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-restrict-cross-namespace-associations.html) but does not actually create the referenced service acount. 

In the Agent/Fleet Server and Beats chart  we actually need to generate a custom service account bound to a role with the use-case-dependent RBAC permissions for the application pods, so that the integrations the user wants to run work and can access the k8s API as needed. 

There are three **problems** with the approach taken: 

1. we coupled it with the cross-ns RBAC feature 
2. we allow to specify a namespace (does not makes sense as you pointed out) and undocumented also custom labels/annotations.
3. we have inconsistent approaches across eck-stack sub-charts

Given that we already released the all three affected charts I don't see how we can now stop supporting these attributes without potentially breaking customer installations.

1. the pointless namespace: its an ugly wart but does not hurt
2. the undocumented annotations/labels overrides. Let's keep them, they don't hurt either
3. the cross-ns coupled with the SA. Violates the principle of least privilege as users have to give the application pods an addtional get permission on Elasticsearch/Kibana  CRD to use the cross-ns RBAC feature that is not needed for the correct operation of the Beats/Agent pods. Again, the paramount concern here is backwards compatibility though. So I would keep it as is for now. Potentially we could introduce `serviceAccountName` in the future that would set the cross-ns SA if specified and otherwise use the `.serviceAccount.name` of the application pod. I don't like that solution very much as it would be very confusing even with extensive documentation. There is another wart that the SA that is being created by the Helm chart is not actually automatically used in the pod template, so the user has to be extra vigilent here and align all the ducks in one row.

**tl;dr**
My suggestion is as follows: 
* accept that we have a different approach with `.serviceAccount.name` for Beats, Agent and Fleet Server
* make sure that in the examples the names line up (you have already done that)
* create follow up issue to untangle the SA/cross-ns RBAC thing potentially in the future and document our decisions.